core/regression_test/regression_test.jl

```diff
@@ -0,0 +1,252 @@
+            end
+        end
+    end
+end
+
+@testitem "regression_ode_solvers_pid_control" begin
+    using SciMLBase: successful_retcode
+    import Arrow
+    using Ribasim
+
+    toml_path = normpath(@__DIR__, "../../generated_testmodels/pid_control/ribasim.toml")
+    @test ispath(toml_path)
+    config = Ribasim.Config(toml_path)
+
+    flow_bytes_bench = read(normpath(@__DIR__, "../../benchmark/pid_control/flow.arrow"))
+    basin_bytes_bench = read(normpath(@__DIR__, "../../benchmark/pid_control/basin.arrow"))
+    flow_bench = Arrow.Table(flow_bytes_bench)
+    basin_bench = Arrow.Table(basin_bytes_bench)
+
+    # TODO "Rosenbrock23" and "Rodas5" solver are resulting unsolvable gradients
+    solver_list = ["QNDF"]
+    sparse_on = [true, false]
+    autodiff_on = [true, false]
+
+    @testset "$solver" for solver in solver_list
+        @testset "sparse density is $sparse_on_off" for sparse_on_off in sparse_on
+            @testset "auto differentiation is $autodiff_on_off" for autodiff_on_off in
+                                                                    autodiff_on
+                config = Ribasim.Config(
+                    toml_path;
+                    solver_algorithm = solver,
+                    solver_sparse = sparse_on_off,
+                    solver_autodiff = autodiff_on_off,
+                )
+                model = Ribasim.run(config)
+                @test model isa Ribasim.Model
+                @test successful_retcode(model)
+                (; p) = model.integrator
+
+                # read all results as bytes first to avoid memory mapping
+                # which can have cleanup issues due to file locking
+                flow_bytes = read(normpath(dirname(toml_path), "results/flow.arrow"))
+                basin_bytes = read(normpath(dirname(toml_path), "results/basin.arrow"))
+
+                flow = Arrow.Table(flow_bytes)
+                basin = Arrow.Table(basin_bytes)
+
+                # Testbench for flow.arrow
+                @test flow.time == flow_bench.time
+                @test flow.edge_id == flow_bench.edge_id
+                @test flow.from_node_type == flow_bench.from_node_type
+                @test flow.from_node_id == flow_bench.from_node_id
+                @test flow.to_node_type == flow_bench.to_node_type
+                @test flow.to_node_id == flow_bench.to_node_id
+                @test all(q -> abs(q) < 0.01, flow.flow_rate - flow_bench.flow_rate)
+
+                # Testbench for basin.arrow
+                @test basin.time == basin_bench.time
+                @test basin.node_id == basin_bench.node_id
+                @test all(q -> abs(q) < 100.0, basin.storage - basin_bench.storage)
+                @test all(q -> abs(q) < 0.5, basin.level - basin_bench.level)
+                @test all(q -> abs(q) < 1e-3, basin.balance_error)
+                @test all(q -> abs(q) < 2.5, basin.relative_error)
```

Comment:
```suggestion
```

Let's leave this out, I think we have to change the way we calculate this. 2.5 is huge.